src/PostgREST/App.hs

```diff
@@ -207,9 +174,11 @@ app dbstructure conf authenticator reqBody dbrole req =
         then do
           let call = B.Stmt "select " V.empty True <>
                 asJson (callProc qi $ fromMaybe M.empty (decode reqBody))
-          body :: Maybe (Identity Text) <- H.maybeEx call
+          bodyJson :: Maybe (Identity Value) <- H.maybeEx call
           return $ responseLBS status200 [jsonH]
-            (cs $ fromMaybe "[]" $ runIdentity <$> body)
+                (if hasPrefer "return=jwt"
```

Comment:
Clever! I see how the user can create whatever claims they need to issue in a stored proc. In production you would probably want to rate limit this endpoint because it provides a way to generate lots of encrypted values and associated plaintext values to help an attacker crack the JWT secret. Not sure how many messages someone would need to do that.